dataregistry-extension/src/csvviewer/data_grid.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license BSD-3-Clause
+ *
+ * Copyright (c) 2019 Project Jupyter Contributors.
+ * Distributed under the terms of the 3-Clause BSD License.
+ */
+
+import { Observable, Subscription } from "rxjs";
+import { DataGrid } from "@phosphor/datagrid";
+import { Message } from "@phosphor/messaging";
+import { DSVModel } from "@jupyterlab/csvviewer";
+
+/**
+ * Class to support the conversion from CSV data to `DataGrid` data.
+ *
+ * @private
+ */
+class CSVDataGrid extends DataGrid {
+  private _data: Observable<string>;
+  private _subscription?: Subscription;
+
+  /**
+   * Constructor.
+   *
+   * @param data - CSV data as an observable
+   * @returns class instance
+   */
+  constructor(data: Observable<string>) {
+    super();
+    this._data = data;
+    this.headerVisibility = "all";
+  }
+
+  /**
+   * Callback invoked upon receiving a `'before-attach'` message.
+   *
+   * @param msg - message
+   */
+  protected onBeforeAttach(msg: Message) {
+    const self = this;
+    this._subscription = this._data.subscribe(onData);
+    super.onBeforeAttach(msg);
+
+    /**
+     * Callback invoked upon changes to the CSV data.
+     *
+     * @private
+     * @param data - CSV data
+     */
+    function onData(data: string) {
+      if (self.model) {
+        (self.model as DSVModel).dispose();
+      }
+      self.model = new DSVModel({
+        "data": data,
+        "delimiter": ","
+      });
+    }
+  }
+
+  /**
+   * Callback invoked upon receiving a `'before-detach'` message.
+   *
+   * @param msg - message
+   */
+  protected onBeforeDetach(msg: Message) {
+    this._subscription && this._subscription.unsubscribe();
+    super.onBeforeDetach(msg);
+  }
+}
+
+/**
+ * Exports.
+ */
+export { CSVDataGrid };
+export default CSVDataGrid;
```

Comment:
I don't think. we need a default export

Reply:
In general, my preference would be that every file have **only** a default export and no named exports for reasons mentioned earlier. However, I mainly included named exports because, with few exceptions, named exports are the norm in this project.
​
This has been debated somewhat in the Node.js community where some developers do both in order to allow users to pick whichever style they prefer and whichever style matches their project conventions.